src/coprocessor/statistics/fmsketch.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// FIXME: remove following later
+#![allow(dead_code)]
+
+use std::collections::HashSet;
+use byteorder::{ByteOrder, LittleEndian};
+use murmur3::murmur3_x64_128;
+use tipb::analyze;
+
+/// `FMSketch` is used to count the approximate number
+/// of distinct elements in multiset.
+pub struct FMSketch {
```

Comment:
Please add a link to https://en.wikipedia.org/wiki/Flajolet%E2%80%93Martin_algorithm.